pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -186,6 +190,30 @@ func maybeSetHashPolicy(destinationRule *networking.DestinationRule, tcpProxy *t
 	}
 }
 
+func maybeSetTunnelingConfig(destRule *networking.DestinationRule, tcpProxy *tcp.TcpProxy, destinationHosts []string, subsetName string) {
+	var tunnelingSettings *networking.TrafficPolicy_TunnelingSettings
+	if subsetName != "" {
+		for _, s := range destRule.GetSubsets() {
```

Comment:
See my other comments - I think we should NOT use DR for configuring this, but something like auto-MTLS or the port name, etc.

Support for POST, CONNECT is a characteristic of the server and deployment - and may be change if user is replacing the proxy or depending on where the app is deployed. User should not have to know.

The larger reason for this is that we'll use same mechanism when moving to 'better transport security/hbone' - where CONNECT will be used for the TCP streams between workloads.

Reply:
I don't agree, as DR is specifically used to configure client connections to upstreams.  While auto detection of the proxy's protocol may be useful, I think there is a use case for manually configuring this for specific hosts and/or endpoints.  For example, if you are integrating with an external service which is not running a proxy, but needs to be accessed through a forward proxy, this supports that use case nicely.  It also does not prevent other refinements or optimizations, e.g. using labels, global settings, etc.

Reply:
> See my other comments - I think we should NOT use DR for configuring this, but something like auto-MTLS or the port name, etc.

Ok, let's consider enabling tunneling based on port name. How would you route traffic to a proper forward proxy relying only on a port name or a port setting?